### PR TITLE
fix(condo): DOMA-3981 add tests for syncUser.js

### PR DESCRIPTION
**Pull request by ekabardinsky**

### CODE REVIEW

File: apps/condo/domains/user/integration/sberid/sync/syncUser.js [hunk 1]
```diff
@@ -66,6 +66,7 @@ const syncUser = async ({ context, userInfo, userType }) => {
     const userIdentities = await UserExternalIdentity.getAll(context, {
         identityType: SBER_ID_IDP_TYPE,
         identityId: userInfo.id,
+        deletedAt: null,
```
Comment (pahaz) [inline apps/condo/domains/user/integration/sberid/sync/syncUser.js:69]: It looks strange. Can we add a todo to remove this parameter? We should by default have only not deleted objects

**Review by pahaz — APPROVED** [review]

File: apps/condo/domains/user/integration/sberid/sync/syncUser.spec.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        // assert id of user
+        expect(id).toEqual(existingUser.id)
+
+        // assert count of external identities
+        const identities = await UserExternalIdentityApi.getAll(context, {
+            user: { id },
+            identityType: SBER_ID_IDP_TYPE,
+        })
+
+        expect(identities.length).toEqual(1)
```
Comment (sitozzz) [inline apps/condo/domains/user/integration/sberid/sync/syncUser.spec.js:126]:
```suggestion
        expect(identities).toHaveLength(1)
```

File: apps/condo/domains/user/integration/sberid/sync/syncUser.spec.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        expect(id).toEqual(existingUser.id)
+
+        // assert user external identity
+        const [ checkedIdentity ] = await UserExternalIdentityApi.getAll(context, {
+            identityId,
+            identityType: SBER_ID_IDP_TYPE,
+        })
+        expect(checkedIdentity).toBeDefined()
+        expect(checkedIdentity.user).toBeDefined()
+        expect(checkedIdentity.user.id).toEqual(id)
```
Comment (sitozzz) [inline apps/condo/domains/user/integration/sberid/sync/syncUser.spec.js:92]: You can use `toHaveProperty` syntax to match object properties value

File: apps/condo/domains/user/integration/sberid/sync/syncUser.spec.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        const userInfo = mockUserInfo(identityId)
+        const context = await keystone.createContext({ skipAccessControl: true })
+
+        // act
+        const { id } = await syncUser({ context, userInfo, userType: RESIDENT })
+
+        // assertions
+        // assert created user
+        expect(id).toBeDefined()
+        const [ checkUser ] = await getItems({
```
[the remark this answers is not in the document]
Comment (sitozzz) [reply apps/condo/domains/user/integration/sberid/sync/syncUser.spec.js:44]: As far as I know it return all fields that you need to check at this test

File: apps/condo/domains/user/integration/sberid/sync/syncUser.js (hunk 1 above)
Comment (ekabardinsky) [reply apps/condo/domains/user/integration/sberid/sync/syncUser.js:69]: ok

File: apps/condo/domains/user/integration/sberid/sync/syncUser.spec.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    issuer: 'testIssuer',
+    email: faker.internet.email(),
+    phoneNumber: phone || faker.phone.phoneNumber('+792########'),
+})
+
+describe('syncUser from SBBOL', () => {
+    setFakeClientMode(index)
+
+    it('should create user', async () => {
+        // assemble
```
Comment (AntonAL) [inline apps/condo/domains/user/integration/sberid/sync/syncUser.spec.js:33]: This comment does not provides additional information. It's clear that before calling a test subject, we are assembling something
```suggestion
```

File: apps/condo/domains/user/integration/sberid/sync/syncUser.spec.js (hunk 2 above)
Comment (ekabardinsky) [reply apps/condo/domains/user/integration/sberid/sync/syncUser.spec.js:126]: ok

File: apps/condo/domains/user/integration/sberid/sync/syncUser.js [hunk 6]
```diff
@@ -66,6 +66,8 @@ const syncUser = async ({ context, userInfo, userType }) => {
     const userIdentities = await UserExternalIdentity.getAll(context, {
         identityType: SBER_ID_IDP_TYPE,
         identityId: userInfo.id,
+        // TODO remove this parameter. We should by default have only not deleted objects
```
[the remark this answers is not in the document]
Comment (ekabardinsky) [reply apps/condo/domains/user/integration/sberid/sync/syncUser.js:69]: ok

File: apps/condo/domains/user/integration/sberid/sync/syncUser.spec.js [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+    phoneNumber: phone || faker.phone.phoneNumber('+792########'),
+})
+
+describe('syncUser from SBBOL', () => {
+    setFakeClientMode(index)
+
+    it('should create user', async () => {
+        const identityId = uuid()
+        const userInfo = mockUserInfo(identityId)
+        const context = await keystone.createContext({ skipAccessControl: true })
```
[the remark this answers is not in the document]
Comment (ekabardinsky) [reply apps/condo/domains/user/integration/sberid/sync/syncUser.spec.js:35]: ok

File: apps/condo/domains/user/integration/sberid/sync/syncUser.spec.js [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+
+        // act
+        const { id } = await syncUser({ context, userInfo, userType: RESIDENT })
+
+        // assertions
+        // assert created user
+        expect(id).toBeDefined()
+        const [ checkUser ] = await getItems({
```
[the remark this answers is not in the document]
Comment (ekabardinsky) [reply apps/condo/domains/user/integration/sberid/sync/syncUser.spec.js]: replaced

**Review by VKislov — APPROVED** [review]

**Pull request merged**